### ISSUE TITLE
plotRGB()  Main and axes do not work again.

### ISSUE THREAD

**Issue by Rapsodia86** [rspatial/terra#528]
Main and axes do not work again. 
I have a dev version: terra 1.5.14 
b <- rast(system.file("ex/logo.tif", package="terra"))   
plotRGB(b,main="IMAGE",axes=TRUE)
and it plots me RGB but no title and no axes.

_Originally posted by @Rapsodia86 in https://github.com/rspatial/terra/issues/85#issuecomment-1030155002_

**Comment by rhijmans** [comment]
The margins in `plotRGB` are zero. So if you want to see the axes and main, you need to set them. E.g. 

```
plotRGB(b,main="IMAGE",axes=TRUE, mar=4)
box()
```

**Comment by Rapsodia86** [comment]
Indeed, sorry for that! 
Any chance to manipulate with theme to match plot() output? e.g. border and axes labels vertical? The same about changing the font?

**Comment by rhijmans** [comment]
You can do 

```
x = RGB2col(b)
``` 

To create a SpatRaster with a colortable (this is under-documented, and you can ignore the warning about col2RGB).
And then use 

```
plot(x)
```

**Comment by Rapsodia86** [comment]
I am getting there , but now have some trouble with the stretch. I do operate on MODIS NDVI without the scale factor  applied, so my values are from 0 to 10000 (as signed INT16). 
So, to you use your approach I have to normalize data to 255 first . PlotRGB() does not need this. Here is the approach: 
(The normalization function is taken from your example at: https://stackoverflow.com/questions/70532643/normalized-values-in-rgb-raster-using-a-function )

library(terra)
#terra 1.5.14
r <- rast("C:/Test/mod_apr.tif")
#> r
#class       : SpatRaster 
#dimensions  : 1200, 1200, 1  (nrow, ncol, nlyr)
#resolution  : 926.6254, 926.6254  (x, y)
#extent      : -7783654, -6671703, 4447802, 5559753  (xmin, xmax, ymin, ymax)
#coord. ref. : +proj=sinu +lon_0=0 +x_0=0 +y_0=0 +R=6371007.181 +units=m +no_defs 
#source      : mod_apr.tif 
#name        : MOD13A3_NDVI3_A2016092_h11v04 
#min value   :                             0 
#max value   :                         10000 

g <- rast("C:/Test/mod_may.tif")
b <- rast("C:/Test/mod_jun.tif")

stc <- c(r,g,b)
plotRGB(stc, 1, 2, 3, stretch='lin',colNA="black", mar=5, axes = TRUE)
![image](https://user-images.githubusercontent.com/55965799/152585672-8b1dd079-5612-4195-87b2-5eea1a77109a.png)

norfun2 <- function(x) {
  rmean <- unlist(global(x, "mean", na.rm=TRUE))
  rsd <- unlist(global(x, "sd", na.rm=TRUE))
  bstd <- (x-rmean)/rsd
  brng <- t(as.matrix(global(bstd, "range", na.rm=TRUE)))
  as.vector((255 / diff(brng))) * (bstd - brng[1,])
}
R <- norfun2(r)  #I know it can be run across all layers at once
#class       : SpatRaster 
#dimensions  : 1200, 1200, 1  (nrow, ncol, nlyr)
#resolution  : 926.6254, 926.6254  (x, y)
#extent      : -7783654, -6671703, 4447802, 5559753  (xmin, xmax, ymin, ymax)
#coord. ref. : +proj=sinu +lon_0=0 +x_0=0 +y_0=0 +R=6371007.181 +units=m +no_defs 
#source      : memory 
#name        : MOD13A3_NDVI3_A2016092_h11v04 
#min value   :                             0 
#max value   :                           255 

G <- norfun2(g)
B <- norfun2(b)

stc2 <- c(R,G,B)
RGB(stc2) <- c(1,2,3)
x <- colorize(stc2, "col", stretch="hist")
y <- colorize(stc2, "col", stretch="lin")
z <- colorize(stc2, "col")

#All plots below are the same

plot(x,colNA="black") 
![image](https://user-images.githubusercontent.com/55965799/152585759-9ec42f4c-d175-4199-a900-104b5adda0cc.png)
plot(y,colNA="black")
![image](https://user-images.githubusercontent.com/55965799/152585874-5b5604bc-cb90-4317-84e6-1ba005378fe1.png)
plot(z,colNA="black")
![image](https://user-images.githubusercontent.com/55965799/152585916-8d07b90e-e419-4179-b673-1c5a576724b1.png)

What  am I missing to get the plot like using plotRGB() function?

Thanks a lot! 
[mod_data_test_plot_terra.zip](https://github.com/rspatial/terra/files/8005508/mod_data_test_plot_terra.zip)

**Comment by Rapsodia86** [comment]
One more thing. When I display those images in e.g. ENVI as RGB, then for pixels where only one (or two) band has a value (other two /or one NA) then I still get a 'color' (see in red square) which is a good information when create false color composite to see the differences. Would that approach be possible in plotRGB() or plot(). Below an example:
![image](https://user-images.githubusercontent.com/55965799/152587578-dd87816c-e3ed-4d1f-9785-84856789c610.png)

**Comment by rhijmans** [comment]
The `stretch` argument in `colorize` was ignored.  That should now be fixed. You can also do it yourself with `x <- stretch(x, qmin=0.02, qmax=0.98)`

**Comment by rhijmans** [comment]
> for pixels where only one (or two) band has a value (other two /or one NA) then I still get a 'color' 

I suppose there could be an argument `NAasZero=FALSE` to treat NA as zero (you can try with your data to see if that gives you the desired output, with something like `xx <- ifel(is.na(x), 0, x) `

**Comment by Rapsodia86** [comment]
Yes! Worked perfect! Thank you very much! 

stc2 <- c(R,G,B)
stc2 <- stretch(stc2, minq=0.02, maxq=0.98) #it is maxq, not qmax ;) 
stc2 <- ifel(is.na(stc2), 0, stc2)

RGB(stc2) <- c(1,2,3)
x <- colorize(stc2, "col")
![image](https://user-images.githubusercontent.com/55965799/152591246-ba860933-7b09-475b-8e33-431d640f2f09.png)

**Comment by rhijmans** [comment]
Great, you can now do: 

```
library(terra)
r <- rast("mod_apr.tif")
g <- rast("mod_may.tif")
b <- rast("mod_jun.tif")

stc <- c(r,g,b)
# perhaps take a sample, as the file is very large (this will be done by plot anyway)
stc <- spatSample(stc, 500000, "regular", as.raster=TRUE)

x <- colorize(stc, value=1:3, "col", stretch='lin', NAzero=TRUE)
plot(x)
```

**Comment by Rapsodia86** [comment]
Is this already implemented in terra 1.5.18?

**Comment by rhijmans** [comment]
yes, it is. But if you are using the R-universe installer you have to wait till it builds terra again. I pushed this to github about 30 minutes ago.  The last build is from about 1 hour ago: https://r-universe.dev/organizations/ 
But it will be built again soon (I do not know when and have no control over that).
